### PR TITLE
Feature/wagtail2

### LINKED ISSUE #169: support for wagtail 2?
I'm running wagtail 2. Is support for wagtail 2 on the roadmap?

Problems include the fact the import paths for wagtail 2 have changed since wagtail 1. i.e.,

`from wagtail.wagtailcore.models import Page` is now `from wagtail.core.models import Page`

### PR DESCRIPTION
**Pull request by stuaxo**
---
Added by @dmarcelino:
This PR fixes #169

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +34,9 @@ matrix:
     env: WAGTAIL="wagtail>=1.5,<1.6"
   - python: "3.5"
     env: WAGTAIL="wagtail>=1.4,<1.5"
+  exclude:
```
Comment (dmarcelino) [inline .travis.yml]: With the current config I don't think this exclude is doing anything, since wagtail 2.0 is only being ran against Python 3.6.

Comment (stuaxo) [reply .travis.yml]: OK, will try removing that.

Main thing i seem to be hitting now is some django-2 issues...  I had to guess what was going on with the migrations (the call_command), and now there is a weird error about clone (see travis).

**Comment by stuaxo** [comment]
Looking into this -

https://travis-ci.org/infoportugal/wagtail-modeltranslation/jobs/347276862#L631

@dmarcelino  - any ideas?

**Comment by dmarcelino** [comment]
Hi @stuaxo, great effort, thanks!

Regarding `travis.yml`, your PR is adding 3 new jobs. I don't think we need to cover that many scenarios, one extra job should be enough. I suggest `travis.yml` should begin with:

```yaml
language: python
python:
  - "3.6"
env:
  - WAGTAIL="wagtail>=2.0,<2.1" DB=sqlite
matrix:
  include:
  # Latest Wagtail version
  - env: WAGTAIL="wagtail>=2.0,<2.1" DB=sqlite
  - env: WAGTAIL="wagtail>=2.0,<2.1" DB=postgres
  - env: WAGTAIL="wagtail>=2.0,<2.1" DB=mysql
  - python: "3.5"
  - python: "3.4"
  # Past Wagtail versions
  - python: "3.6"
    env: WAGTAIL="wagtail>=1.13,<1.14" 
  - python: "2.7"  # Wagtail 1.13 was the latest tested against 2.7
    env: WAGTAIL="wagtail>=1.13,<1.14" 
  - python: "3.6"
    env: WAGTAIL="wagtail>=1.12,<1.13"
    # ...
```
This ensures we test with the latest wagtail version against python 3.6, 3.5 and 3.4 as well as sqlite, postgres and mysql. For older versions we do less tests to keep the travis build snappy.

We should also drop anything below Wagtail 1.8 (penultimate LTS) at some point.

**Comment by stuaxo** [comment]
Great, that is sensible with travis.

**Comment by dmarcelino** [comment]
That was a fast change! 🙂 

But you've missed the part where I moved `python: "2.7"` and `WAGTAIL="wagtail>=1.13,<1.14" ` from the `# Latest Wagtail version` section to the `# Past Wagtail versions` section.

File: wagtail_modeltranslation/tests/tests.py [hunk 2]
```diff
@@ -94,15 +98,27 @@ def setUpClass(cls):
 
                 # 5. makemigrations
                 from django.db import connections, DEFAULT_DB_ALIAS
-                call_command('makemigrations', verbosity=2, interactive=False,
-                             database=connections[DEFAULT_DB_ALIAS].alias)
+                if VERSION[0] < 2:
```
Comment (dmarcelino) [inline wagtail_modeltranslation/tests/tests.py]: I'm curious about the changes to `call_command` invocation, has Wagtail 2.0 changed the API?

Comment (stuaxo) [reply wagtail_modeltranslation/tests/tests.py]: I need to put this inside a check for django.

Later versions of django have changed the parameters that makemigrations and migrate support.   Have to say, I was kind of guessing what would work.

You can probably see some earlier travis builds where it was complaining about unsupported parameters.

Comment (dmarcelino) [reply wagtail_modeltranslation/tests/tests.py]: I see.

**Comment by stuaxo** [comment]
Oops, apologies on my bad cut n pasting with .travis :-/

Here are the errors I was trying to work around with call_command
```
TypeError: Unknown option(s) for migrate command: load_initial_data, migrate. Valid options are: app_label, database, fake, fake_initial, help, interactive, migration_name, no_color, no_input, pythonpath, run_syncdb, settings, skip_checks, stderr, stdout, traceback, verbosity, version.
```

```
TypeError: Unknown option(s) for migrate command: migrate. Valid options are: app_label, database, fake, fake_initial, help, interactive, migration_name, no_color, no_input, pythonpath, run_syncdb, settings, skip_checks, stderr, stdout, traceback, verbosity, version.
```

Passing in the database doesn't seem to be an option any more. 
run_syncdb isn't a thing, and load_initial_data as well (which was False anyway).
I'm not sure if adding fake=True was the right thing to do though.

**Comment by dmarcelino** [comment]
Have you tried running the tests on your local machine in debug mode?

**Comment by stuaxo** [comment]
How do I set debug mode ?

The tests seem to pass locally -


```
$  python runtests.py 
Creating test database for alias 'default' ('file:memorydb_default?mode=memory&cache=shared')...
Operations to perform:
  Synchronize unmigrated apps: api, rest_framework, wagtail_modeltranslation, wagtail_modeltranslation_makemigrations, wagtailsettings, wagtailsites, wagtailsnippets
  Apply all migrations: auth, contenttypes, taggit, wagtailadmin, wagtailcore, wagtaildocs, wagtailembeds, wagtailforms, wagtailimages, wagtailredirects, wagtailsearch, wagtailusers
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_auto_20180129_1758... OK
  Applying taggit.0001_initial... OK
  Applying taggit.0002_auto_20150616_2121... OK
  Applying wagtailcore.0001_squashed_0016_change_page_url_path_to_text_field... OK
  Applying wagtailcore.0017_change_edit_page_permission_description... OK
  Applying wagtailcore.0018_pagerevision_submitted_for_moderation_index... OK
  Applying wagtailcore.0019_verbose_names_cleanup... OK
  Applying wagtailcore.0020_add_index_on_page_first_published_at... OK
  Applying wagtailcore.0021_capitalizeverbose... OK
  Applying wagtailcore.0022_add_site_name... OK
  Applying wagtailcore.0023_alter_page_revision_on_delete_behaviour... OK
  Applying wagtailcore.0024_collection... OK
  Applying wagtailcore.0025_collection_initial_data... OK
  Applying wagtailcore.0026_group_collection_permission... OK
  Applying wagtailadmin.0001_create_admin_access_permissions... OK
  Applying wagtailcore.0027_fix_collection_path_collation... OK
  Applying wagtailcore.0024_alter_page_content_type_on_delete_behaviour... OK
  Applying wagtailcore.0028_merge... OK
  Applying wagtailcore.0029_unicode_slugfield_dj19... OK
  Applying wagtailcore.0030_index_on_pagerevision_created_at... OK
  Applying wagtailcore.0031_add_page_view_restriction_types... OK
  Applying wagtailcore.0032_add_bulk_delete_page_permission... OK
  Applying wagtailcore.0033_remove_golive_expiry_help_text... OK
  Applying wagtailcore.0034_page_live_revision... OK
  Applying wagtailcore.0035_page_last_published_at... OK
  Applying wagtailcore.0036_populate_page_last_published_at... OK
  Applying wagtailcore.0037_set_page_owner_editable... OK
  Applying wagtailcore.0038_make_first_published_at_editable... OK
  Applying wagtailcore.0039_collectionviewrestriction... OK
  Applying wagtailcore.0040_page_draft_title... OK
  Applying wagtaildocs.0001_initial... OK
  Applying wagtaildocs.0002_initial_data... OK
  Applying wagtaildocs.0003_add_verbose_names... OK
  Applying wagtaildocs.0004_capitalizeverbose... OK
  Applying wagtaildocs.0005_document_collection... OK
  Applying wagtaildocs.0006_copy_document_permissions_to_collections... OK
  Applying wagtaildocs.0005_alter_uploaded_by_user_on_delete_action... OK
  Applying wagtaildocs.0007_merge... OK
  Applying wagtailembeds.0001_initial... OK
  Applying wagtailembeds.0002_add_verbose_names... OK
  Applying wagtailembeds.0003_capitalizeverbose... OK
  Applying wagtailforms.0001_initial... OK
  Applying wagtailforms.0002_add_verbose_names... OK
  Applying wagtailforms.0003_capitalizeverbose... OK
  Applying wagtailimages.0001_initial... OK
  Applying wagtailimages.0002_initial_data... OK
  Applying wagtailimages.0003_fix_focal_point_fields... OK
  Applying wagtailimages.0004_make_focal_point_key_not_nullable... OK
  Applying wagtailimages.0005_make_filter_spec_unique... OK
  Applying wagtailimages.0006_add_verbose_names... OK
  Applying wagtailimages.0007_image_file_size... OK
  Applying wagtailimages.0008_image_created_at_index... OK
  Applying wagtailimages.0009_capitalizeverbose... OK
  Applying wagtailimages.0010_change_on_delete_behaviour... OK
  Applying wagtailimages.0011_image_collection... OK
  Applying wagtailimages.0012_copy_image_permissions_to_collections... OK
  Applying wagtailimages.0013_make_rendition_upload_callable... OK
  Applying wagtailimages.0014_add_filter_spec_field... OK
  Applying wagtailimages.0015_fill_filter_spec_field... OK
  Applying wagtailimages.0016_deprecate_rendition_filter_relation... OK
  Applying wagtailimages.0017_reduce_focal_point_key_max_length... OK
  Applying wagtailimages.0018_remove_rendition_filter... OK
  Applying wagtailimages.0019_delete_filter... OK
  Applying wagtailredirects.0001_initial... OK
  Applying wagtailredirects.0002_add_verbose_names... OK
  Applying wagtailredirects.0003_make_site_field_editable... OK
  Applying wagtailredirects.0004_set_unique_on_path_and_site... OK
  Applying wagtailredirects.0005_capitalizeverbose... OK
  Applying wagtailsearch.0001_initial... OK
  Applying wagtailsearch.0002_add_verbose_names... OK
  Applying wagtailsearch.0003_remove_editors_pick... OK
  Applying wagtailusers.0001_initial... OK
  Applying wagtailusers.0002_add_verbose_name_on_userprofile... OK
  Applying wagtailusers.0003_add_verbose_names... OK
  Applying wagtailusers.0004_capitalizeverbose... OK
  Applying wagtailusers.0005_make_related_name_wagtail_specific... OK
  Applying wagtailusers.0006_userprofile_prefered_language... OK
System check identified no issues (0 silenced).
No changes detected
Missing languages in "search_description" field from "wagtailcore.page" model: de, en

SQL to synchronize "wagtailcore.page" schema:
   ALTER TABLE "wagtailcore_page" ADD COLUMN "search_description_de" text;
   ALTER TABLE "wagtailcore_page" ADD COLUMN "search_description_en" text;
Executing SQL...
Done
Missing languages in "seo_title" field from "wagtailcore.page" model: de, en

SQL to synchronize "wagtailcore.page" schema:
   ALTER TABLE "wagtailcore_page" ADD COLUMN "seo_title_de" varchar(255);
   ALTER TABLE "wagtailcore_page" ADD COLUMN "seo_title_en" varchar(255);
Executing SQL...
Done
Missing languages in "url_path" field from "wagtailcore.page" model: de, en

SQL to synchronize "wagtailcore.page" schema:
   ALTER TABLE "wagtailcore_page" ADD COLUMN "url_path_de" text;
   ALTER TABLE "wagtailcore_page" ADD COLUMN "url_path_en" text;
Executing SQL...
Done
Missing languages in "slug" field from "wagtailcore.page" model: de, en

SQL to synchronize "wagtailcore.page" schema:
   ALTER TABLE "wagtailcore_page" ADD COLUMN "slug_de" varchar(255);
   ALTER TABLE "wagtailcore_page" ADD COLUMN "slug_en" varchar(255);
Executing SQL...
Done
Missing languages in "title" field from "wagtailcore.page" model: de, en

SQL to synchronize "wagtailcore.page" schema:
   ALTER TABLE "wagtailcore_page" ADD COLUMN "title_de" varchar(255);
   ALTER TABLE "wagtailcore_page" ADD COLUMN "title_en" varchar(255);
Executing SQL...
Done
test_duplicate_slug (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_fetch_translation_records (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_get_url_parts (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_page_fields (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_page_form (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_page_patching (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_relative_url (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_request_routing (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_searchfield_patching (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_set_translation_url_paths_command (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_set_url_path (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_set_url_path_non_translated_descendants (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_slugurl_trans (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_snippet_fields (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_snippet_form (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_snippet_patching (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_streamfield_fallback (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok
test_url (wagtail_modeltranslation.tests.tests.WagtailModeltranslationTest) ... ok

----------------------------------------------------------------------
Ran 18 tests in 3.107s

OK
Destroying test database for alias 'default' ('file:memorydb_default?mode=memory&cache=shared')...

```

**Comment by stuaxo** [comment]
I'll try updating wagtail, currently wagtail.VERSION is ```VERSION:  (2, 0, 0, 'beta', 1)```

**Comment by stuaxo** [comment]
OK.. reproduced it.

I was using django 1.11.10  with django 2 - the tests pass there.

**Comment by dmarcelino** [comment]
Try using wagtail 2.0 and django 2.0 in your local test environment.

In regard to debugging, if you are using pyCharm, https://www.jetbrains.com/help/pycharm/debugging.html is a good starting point.

**Comment by stuaxo** [comment]
I've never used the pycharm debugger, will give it a go.   Now this works on 1.11.* I'm going to see do some testing with that.

Just trying to add this to an existing project and getting this error when running makemigrations - any ideas? 

```
no module named 'wagtail_modeltranslation.models'
```

**Comment by stuaxo** [comment]
So I guess wagtail2 != django2.

**Comment by stuaxo** [comment]
I think I'll separate out the django 2 stuff from wt2.

Since wt2 doesn't need django 2.

**Comment by dmarcelino** [comment]
Hey @stuaxo,

I haven't had the time to study Django 2.0 changes, but the problem you're seeing may be related to the following change in Django: [call_command() validates the options it receives](https://docs.djangoproject.com/en/2.0/releases/2.0/#call-command-validates-the-options-it-receives).

Also, it's a good idea to look at `django-modeltranslation` to see how they are dealing with this (our test suite is inspired on theirs), namely take a look at PR deschler/django-modeltranslation#451.

Cheers

**Comment by stuaxo** [comment]
Great, that looks like their fix is is the right one + can replace the bits where I was guessing.

**Comment by stuaxo** [comment]
OK, so it's down to the _rewrite issue now.
This is something triggering in django_treebeard.    

I might try testing locally with the django-modeltranslations from git in case there are some fixes here.

+ if that fails, I guess asking for help from the django-modeltranslation project might be worthwhile.

**Comment by stuaxo** [comment]
OK, tests pass if you install django-modeltranslation from git... maybe we can persuade them to do a release ?

**Comment by dmarcelino** [comment]
@stuaxo, I've split the wagtail2 changes from the django2 ones and opened a new PR, check it out: #185.

**Comment by dmarcelino** [comment]
Hey @stuaxo, since PR #185 has been merged I'm closing this one. Thanks for your contributions! For Django 2 support please submit a new PR.

**Comment by stuaxo** [comment]
Fantastic :+1:

Will see what I can do wrt django 2, though the project I'm currently working on is not using it, so I might not catch everything.

**Comment by dmarcelino** [comment]
Until a new version of `django-modeltranslation` comes out there isn't much we can do except, perhaps, to help them fix the 2 broken tests, see https://github.com/deschler/django-modeltranslation/issues/436#issuecomment-371594718

Thanks again!

**Pull request closed without merging**